frontend/src/components/SearchForm.js

```diff
@@ -14,21 +14,42 @@ const SearchForm = () => (
 
             <div class="radio-bar-container">
                 <div class="radio-choice-container">
-                    <input type="radio" id="beginner" name="select" value="beginner" />
+                    <input 
+                        type="radio" 
+                        id="beginner" 
+                        name="select" 
+                        value="beginner" 
+                        checked={radioDifficulty === 'beginner'} 
+                        onClick={() => setRadioDifficulty('beginner')}
+                    />
                     <label for="beginner">
```

Comment:
This is not too different from the reason given above, but your  console should contain this error too. Use `htmlFor` rather than `for` in label elements in react. Please, find and correct every occurence.

Reply:
This is new for me, will change it.